TAO/TAO_IDL/be/be_visitor_field/field_ch.cpp

```diff
@@ -380,6 +381,92 @@ be_visitor_field_ch::visit_sequence (be_sequence *node)
   return 0;
 }
 
+int
+be_visitor_field_ch::visit_map (be_map *node)
+{
+  TAO_OutStream *os = this->ctx_->stream ();
+  be_type *bt = nullptr;
+  if (this->ctx_->alias ())
+    {
+      bt = this->ctx_->alias ();
+    }
+  else
+    {
+      bt = node;
+    }
+
+  if (!this->ctx_->alias ()
+      && node->is_child (this->ctx_->scope ()->decl ()))
+    {
+      // Put the field node into the (anonymous) sequence node, to be
+      // used later for unique name generation.
+      be_field *member_node =
+        dynamic_cast<be_field*> (this->ctx_->node ());
+      node->field_node (member_node);
+
+      // This was already generated in the corresponding valuetype class.
+      if (this->ctx_->state () != TAO_CodeGen::TAO_VALUETYPE_OBV_CH)
+        {
+          be_visitor_context ctx (*this->ctx_);
+          ctx.node (node);
+
+          // First generate the map declaration.
+          be_visitor_map_ch visitor (&ctx);
+          if (node->accept (&visitor) == -1)
+            {
+              ACE_ERROR_RETURN ((LM_ERROR,
+                                "(%N:%l) be_visitor_field_ch::"
+                                "visit_map - "
+                                "codegen failed\n"),
+                                -1);
+            }
+        }
+
+        // If we are being reused by valutype, this would get generated
+        // in the private section of the OBV_xx class, so we must
+        // generate the typedef for that case elsewhere.
+        AST_Decl::NodeType snt =
+          this->ctx_->scope ()->decl ()->node_type ();
+
+        if (snt != AST_Decl::NT_valuetype && snt != AST_Decl::NT_eventtype)
+          {
+            // Generate the anonymous sequence member typedef.
+            be_decl *bs = this->ctx_->scope ()->decl ();
+
+            TAO_INSERT_COMMENT (os);
+
+            *os << "typedef " << bt->nested_type_name (bs)
+                << " _" << this->ctx_->node ()->local_name ()
+                << "_map;" << be_nl;
+          }
+    }
+
+  // ACE_NESTED_CLASS macro generated by nested_type_name
+  // is not necessary in all cases.
+  be_typedef *tdef = dynamic_cast<be_typedef*> (bt);
+
+  // This was a typedefed array.
+  // ACE_NESTED_CLASS macro generated by nested_type_name
```

Comment:
Remove the comments of ACE_NESTED_CLASS, that is old, will do a cleanup on TAO_IDL on master for that